nixos/modules/profiles/macos-builder.nix

```
@@ -93,7 +104,23 @@ in
         };
       });
 
-  system.stateVersion = "22.05";
+  system.stateVersion = lib.mkDefault (throw ''
+    The macOS linux builder should not need a stateVersion to be set, but a module
+    has accessed stateVersion nonetheless.
+    Please inspect the trace of the following command to figure out which module
+    has a dependency on stateVersion.
+
+      nix-instantiate --argstr system x86_64-darwin -A darwin.builder --show-trace
+
+    or
+
+      nix-instantiate --argstr system aarch64-darwin -A darwin.builder --show-trace
```

Does the `system` need to be explicitly specified?

Also, I try to use long option names (e.g. `--attr`) when possible (especially if the user will likely just copy-and-paste the command)

```suggestion
      nix-instantiate --attr darwin.builder --show-trace
```